$MirOS: ports/emulators/qemu/patches/patch-hw_ppc_chrp_c,v 1.2 2005/11/15 12:25:09 tg Exp $
--- hw/ppc_chrp.c.orig	Sun Sep  4 17:11:09 2005
+++ hw/ppc_chrp.c	Tue Nov 15 12:08:12 2005
@@ -355,7 +355,7 @@ static void ppc_chrp_init(int ram_size, 
         kernel_base = KERNEL_LOAD_ADDR;
         /* now we can load the kernel */
         kernel_size = load_image(kernel_filename, phys_ram_base + kernel_base);
-        if (kernel_size < 0) {
+        if ((int32_t)kernel_size < 0) {
             fprintf(stderr, "qemu: could not load kernel '%s'\n", 
                     kernel_filename);
             exit(1);
@@ -365,7 +365,7 @@ static void ppc_chrp_init(int ram_size, 
             initrd_base = INITRD_LOAD_ADDR;
             initrd_size = load_image(initrd_filename,
                                      phys_ram_base + initrd_base);
-            if (initrd_size < 0) {
+            if ((int32_t)initrd_size < 0) {
                 fprintf(stderr, "qemu: could not load initial ram disk '%s'\n", 
                         initrd_filename);
                 exit(1);
@@ -547,10 +547,12 @@ QEMUMachine core99_machine = {
     "mac99",
     "Mac99 based PowerMAC",
     ppc_core99_init,
+    NULL
 };
 
 QEMUMachine heathrow_machine = {
     "g3bw",
     "Heathrow based PowerMAC",
     ppc_heathrow_init,
+    NULL
 };
